tools/ReleaseTools.psm1

```diff
@@ -344,23 +401,29 @@ function New-DraftRelease {
         [ValidateSet([RepoNames])]
         [string]$RepositoryName,
 
-        [Parameter(ValueFromPipeline)]
+        [Parameter()]
         [string[]]$Assets
     )
     $Version = Get-Version -RepositoryName $RepositoryName
     $Changelog = (Get-FirstChangelog -RepositoryName $RepositoryName) -join "`n"
     $ReleaseParams = @{
-        Draft      = $true
         # NOTE: We rely on GitHub to create the tag at that branch.
-        Tag        = "v$Version"
-        Committish = "release/v$Version"
-        Name       = "v$Version"
-        Body       = $ChangeLog
-        PreRelease = [bool]$Version.PreReleaseLabel
-        OwnerName  = "PowerShell"
+        Tag            = "v$Version"
+        Committish     = "release/v$Version"
+        Name           = "v$Version"
+        Body           = $ChangeLog
+        Draft          = $true
+        PreRelease     = [bool]$Version.PreReleaseLabel
+        OwnerName      = "PowerShell"
         RepositoryName = $RepositoryName
+        WhatIf         = $WhatIfPreference
+        Confirm        = $ConfirmPreference
```

Comment:
Actually, @rjmholt this seems to cause `New-GitHubRelease` to _always_ act like `-Confirm` was passed.

Reply:
I think `-Confirm` is a switch but `$ConfirmPreference` is an enum which never takes a falsey value, so it will behave as you describe.

I'm guessing we don't really want to confirm here, but instead are more interested in the WhatIf functionality

Reply:
I forgot `[CmdletBinding(SupportsShouldProcess)]`, that was the issue here 😅